jsdoc/tag-defs/access.js

```diff
@@ -1,13 +1,16 @@
-module.exports = function(accessTagTransform, trimWhitespaceTransform) {
-  var name = 'access'
-  accessTagTransform.addTag(name);
-  accessTagTransform.addValue('private');
-  accessTagTransform.addValue('protected');
-  accessTagTransform.addValue('public');
-
+module.exports = function(createDocMessage) {
+  STANDARD_VALUES = ['private', 'protected', 'public'];
   return {
-    name: name,
-    docProperty: 'access',
-    transforms: [trimWhitespaceTransform, accessTagTransform]
+    name: 'access',
+    transforms: function(doc, tag, value) {
+      var value = value.trim();
+      if (STANDARD_VALUES.indexOf(value) !== -1) {
+        if (doc.tags.getTags(value).length > 0) {
+          throw new Error(createDocMessage('"@access ' + value + '" tag cannot be used with "@' + value + '" tag on the same document', doc));
```

Comment:
@wingedfox - thanks for chiming in.

1. You are right that this does not block that situation.
2. This does not prevent you from doing `@access package-protected` nor does it prevent you from adding your own tagdef for `@package-protected`. What is doesn't do is automatically set the `.packageProtected` property to true if `@access package-protected` is used, nor does it automatically prevent `@access package-protected` and `@package-protected` appearing on the same document.

I think that the best solution is to move all of these checks into a processor that can be configured and disabled if not required, keeping the tags simpler.